services/datamanager/builtin/builtin.go

```diff
@@ -117,6 +119,7 @@ func NewBuiltIn(
 		collectors:                  make(map[componentMethodMetadata]*collectorAndConfig),
 		syncIntervalMins:            0,
 		additionalSyncPaths:         []string{},
```

Comment:
I think it might be cleaner if this is a map of "path": additionalTags

Reply:
What do you mean by instead of having two variables?

Reply:
currently you have it as `additionalSyncPaths` and `additionalTags` separately. But might be better to track them together as one

Reply:
Will look into it!

Reply:
I see this was resolved, but there are still the two separate variables. I would also be partial towards keeping these as a map, since right now there's an implicit index based association between these two arrays

Reply:
Talked to Alexis and we decided it would make the frontend too complicated because the map would need to be modified every time the path was edited.

Reply:
I'm not sure I follow the connection to the FE here, can you elaborate?

Reply:
Pretty sure the backend data structure needs to match the frontend data structure defined in the `types.ts` file. @alexis-wei is this correct?

Reply:
So was originally thinking that if you change the path, then you would have to replace both the key and the values on every input change, but could make it into a map at the end right before making the api call to simplify that procedure. Either way is okay on the FE but a map might be an extra step which is fine.

Reply:
Yeah I'm actually here because I was looking at the FE side of this code. If I understand this correctly `additionalTags` and `additionalSyncPaths` are related arrays in that each sync path is associated with an array of tags? That's why y'all were thinking to do a map? I agree with this since now you'd have to rely on the indexes being the same which means you have to do extra work to keep them in sync which feels fragile. Also are additional paths unique? I imagine users would not typically be using the same paths but yeah that might mess with the map strategy if we don't require them to be unique.

I guess ideally `additionalSyncPaths` should change to a struct with and `name` and `tags` properties but that would introduce a breaking change requiring migration so maybe not in scope

Reply:
no the desired config isn't described anywhere, but I agree that the map would make it easier for users to understand which tags go with which path, I think what Alexis suggested might be best for right now.

Reply:
> Also are additional paths unique? I imagine users would not typically be using the same paths but yeah that might mess with the map strategy if we don't require them to be unique.

I think they should be/sort of need to be unique - it wouldn't really make sense to configure a path to sync multiple times

Reply:
Discussed offline:

- This is a config change, so requires a (brief!) scope
- For now, will work on the assumption that the config will contain a map
- The BE will use a map
- FE stuff being discussed right now

Reply:
As part of this ticket, are we making a scope then?

Reply:
Yeah, we should. Hopefully basically one sentence saying, "here's why, here's the new field in the config"